### PR TITLE
feat: add zpca plugin

### LINKED ISSUE #18: ZPCA scatterplot
**Is your feature request related to a problem? Please describe.**
In the [ZARP](https://git.scicore.unibas.ch/zavolan_group/pipelines/zarp) pipeline, we have the PCA results from zpca as png images and not present in the multiqc report.

**Describe the solution you'd like**
What needs to be done is to add the plots in the multiqc report as scatterplots. So, we need to read zpca outpu tand display it.

### PR DESCRIPTION
**Pull request by krish8484**
## Description

Add plugin that will plot `multiqc` plots for `ZPCA` scores.

Fixes #18 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage

### CODE REVIEW

**Comment by krish8484** [comment]
After running the test analysis on https://github.com/zavolanlab/zpca and obtaining the `PCA.tsv` file, code has been written that takes in this `tsv` file and plots three scatter plots (PC1vPC2, PC1vPC3, PC2vPC3). I have checked the multiqc report with the `png` images(that were generated after running the analysis) , they are similar.
However, after running the analysis in https://github.com/zavolanlab/zpca, we also get a `bargraph` named `scree_plot.png`. Do we need to include this in our multiqc report as well? Currently, the script does not include this part.
Lastly, we must add the description and link for this plugin in the file `modules/zpca/zpca.py`(specified as TO DO). @fgypas @AngryMaciek Please let me know what needs to be added here.

**Comment by fgypas** [comment]
Hi @krish8484  Thank you for your work. Can you please first upload a screenshot of the zpca plot of the report?
Then we can decide whether to show the scree plot or not.

**Comment by krish8484** [comment]
Here is an example. If you wish to see the interactive `html`, it is available at `tests/zpca` in this branch.
mqc_hcplot_bowqigjlns

**Comment by fgypas** [comment]
Thanks a lot @krish8484 
So, normally the X and Y-axis show the explained variance (that is represented in the data frame from the scree plot). 
So, either add this percentage in the X and Y axis or add the scree plot as a barplot. Whatever it's easier for you.
Please, let me know if something is not clear.

File: README.md [hunk 1]
```diff
@@ -13,8 +13,10 @@ To see a miniature example of a MultiQC plugin, visit [example-plugin].
 Custom Plugins in this repository by Zavolan Lab, University of Basel, Switzerland:
 - [ALFA]
 - [TIN-scores]
+- [ZPCA]
 
 [multiqc-plugins]: https://multiqc.info/docs/#multiqc-plugins
 [example-plugin]: https://github.com/MultiQC/example-plugin
 [ALFA]: https://github.com/biocompibens/ALFA
 [TIN-scores]: https://git.scicore.unibas.ch/zavolan_group/tools/tin_score_calculation
```
Comment (AngryMaciek) [inline README.md]: We need to update the link, we moved the repo to GitHub:
https://github.com/zavolanlab/tin-score-calculation

File: tests/zpca/multiqc_data/multiqc_sources.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+Module	Section	Sample Name	Source
```
Comment (AngryMaciek) [inline tests/zpca/multiqc_data/multiqc_sources.txt:1]: I'm sorry, we worked on this some time ago and I forgot things.
What is the purpose of this file? Is this an input to the _zpca_?

File: modules/zpca/zpca.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        """
+        Initializes the parent function of MultiQC and calls
+        relevant functions for the searching of files and the
+        plotting of graphs.
+        """
+
+        super(MultiqcModule, self).__init__(
+            name="ZPCA",
+            anchor="zpca",
+            href="TO ADD",  # TO DO: Add link and description
```
Comment (AngryMaciek) [inline modules/zpca/zpca.py]: todo?

**Review by AngryMaciek — COMMENTED** [review]
The code looks really great! Just some minor changes in the info, please.

File: modules/zpca/zpca.py (hunk 3 above)
Comment (AngryMaciek) [reply modules/zpca/zpca.py]: Ah right, I just saw your question regarding this field.
Well, the link is the GH repo: https://github.com/zavolanlab/zpca
For the description we have to ask @fgypas , as I cannot find anything in his README file.

**Comment by AngryMaciek** [comment]
If the info about explained variance is present in the input files then it should be doable to adjust the plot labels for the interactive _multiqc_-like scatterplots. @krish8484 could you take a look at it?

File: tests/zpca/multiqc_data/multiqc_sources.txt (hunk 2 above)
Comment (krish8484) [reply tests/zpca/multiqc_data/multiqc_sources.txt:1]: This is a file generated by `multiqc`. In fact, all the files in `multiqc_data` have been generated by `multiqc`, so stored it in `tests\` just like in other modules.

**Comment by krish8484** [comment]
Yes, now the script plots the `scree` plot after taking input from `scree.tsv` **and** also adds the `explained variance` in the labels.
Now we can choose, what is better (the plot or the labels) and remove the other. We can also keep both and let the report stay as it is. You can view the example report in `tests/zpca/multiqc_report.html`. Please let me know if anything else needs to be changed.
And thanks for the review @AngryMaciek @fgypas .

**Comment by AngryMaciek** [comment]
@krish8484 
OK, I took a look at the _html_ report.

* The section title: is it _ZPCA_ as in the [issue](https://github.com/zavolanlab/multiqc-plugins/issues/18) or _zpca_ is in the [repo README](https://github.com/zavolanlab/zpca)? @fgypas 
* Section's description - still waiting for the info from @fgypas 
* Since we are operating only on 3 components (and 3 scatterplots therefore) I believe that the separate scree barplot is not necessary. Scree plots are usually used when one has multiple points and would like to estimate the 'elbow'. In our case it would be just duplication of the information from the axes labels so I would remove it not to clog the final report.
* I would adjust the plot's titles: from `PCA components x & y` to `PCA components: x & y`
* I would adjust the scatterplots labels according to the following example: `PC1 - 40.7` --> `PC1 (40.7% variance explained)`

After these are addressed I can merge.

**Comment by fgypas** [comment]
- I agree that the zpca report looks good.
- The title for the section is fine as zpca I think.
- For section description let's add a simple "PCA analysis"
- I agree that we should not show the scree plot and I agree with @AngryMaciek 's proposal

@krish8484 Can you adjust these final changes? Thank you in advance.

**Comment by krish8484** [comment]
Addressed all the changes. :rocket:

**Comment by fgypas** [comment]
Hi @krish8484 
I think @AngryMaciek means to change the labels from `PC1 - 40.7` to `PC1 (40.7% variance explained)`. Not keep both.

**Comment by AngryMaciek** [comment]
Yes, the labels need to be corrected.
Also, as @fgypas wrote above: it is fine as _zpca_, non-capitals.

**Comment by krish8484** [comment]
Yes, sorry for the delay, made the changes now.

**Comment by AngryMaciek** [comment]
Dear @krish8484,
The section is still in capital letters in the test report, could you please take a look?
Maybe it's just for me...<?>

**Comment by krish8484** [comment]
@AngryMaciek have you reinstalled the plugin? It's in lowercase for me, attached a screenshot.
Screenshot from 2021-03-19 23-13-06

**Comment by AngryMaciek** [comment]
I was not installing/reinstalling anything, I just looked at the test _html_.
Maybe it's not updated?

**Comment by krish8484** [comment]
Now, is it updated?

**Comment by AngryMaciek** [comment]
OK, now looks good.
I hate to say it but I think you forgot `%` in the labels. Just add this symbol and we are done here.

**Pull request merged**